scripts/tasks.py

```diff
@@ -220,3 +222,68 @@ def merge(config):
     stream_to_mtz.cmd_report(foms=foms, nshells=task.nshells)
     stream_to_mtz.cmd_get_hkl()
     logger.info(f'Executable written to {stream_to_mtz.tmp_exe}')
+
+def refine_geometry(config, task=None):
+    from btx.diagnostics.geoptimizer import Geoptimizer
+    from btx.misc.shortcuts import fetch_latest, check_file_existence
+    setup = config.setup
+    if task is None:
+        task = config.refine_geometry
+        for var in [task.dx, task.dy, task.dz]:
+            var = tuple([float(elem) for elem in var.split()])
+            var = np.linspace(var[0], var[1], int(var[2]))
```

Comment:
These 3 lines seem superfluous?

Reply:
They're included for running the `refine_geometry` task directly, since in this case the `dx` etc. variables of the refine_geometry portion of the config file need to be parsed. Though there's probably a more efficient way to prase those variables and set the unneeded ones to [0] between the three tasks.